src/cycax/cycad/cycad_side.py

```diff
@@ -11,6 +13,71 @@ def __str__(self):
     def __repr__(self):
         return "Side: " + self.name
 
+    def _location_calc(self, pos: tuple[float, float], sink: float = 0.0) -> tuple[float, float, float]:
+        """Location is calculated for the (x, y) plane using two values and a side.
+
+        Args:
+            pos: (x, y) location of an object which is used to find the location with the side.
+            sink: The object can be sunk bellow the surface of the specified side to make a pocket.
+
+        Returns:
+            The (x, y, z) location of an object.
+        """
+        raise ValueError("_location_calc is Not implimented on" + self.name)
+
+    def _depth_check(self, val: float = None):
+        """Determine the depth of a feature.
+
+        Args:
+            val: Depth of object.
+
+        Raises:
+            ValueError: The side specidfied does not have a depth check method.
+        """
+        raise ValueError("_depth_check is Not implimented on" + self.name)
+
+    def _rotate(self):
+        """Rotate the part by 90 degress by keeping selected side on the same plain."""
+        msg = "_rotate should be defined for each side."
+        raise NotImplementedError(msg)
+
+    def rotate(self, angle: float = 90):
+        """Rotate the part by keeping the selected side on the same plain.
+
+        Rotation is counter clock wise. Negative angles are converted to poistive.
+
+        Args:
+            angle: The angle in degress that the part will be rotates on.
+
+        """
+        if self._parent.assembly is None:
+            # TODO: Think about which exception should be raised, maybe even a custom one.
+            msg = "Not part of an assembly"
+            raise ValueError(msg)
+
+        if angle % 90 != 0:
+            # TODO: Think about which exception should be raised, maybe even a custom one.
+            msg = "Can only rotate in multiples of 90 degrees"
+            raise ValueError(msg)
+
+        while angle < 0:
+            # Change the angle to a positive number.
+            angle = 360 + angle
+
+        for _n in range(int(angle / 90)):
+            self._rotate()
+
+    @property
+    def opposite(self):
```

Comment:
This allow one to get the opposite side of a given side. e.g. part.left.opposite gets you the right side.